packages/governance/src/tests/test_votes.cairo

```diff
@@ -0,0 +1,372 @@
+    // Mint ERC_721_INITIAL_MINT NFTs to DELEGATOR
+    let mut i: u256 = 0;
+    while i < ERC_721_INITIAL_MINT {
+        mock_state.erc721.mint(DELEGATOR(), i);
+        i += 1;
+    };
+    state
+}
+
+fn setup_erc20_votes() -> ERC20ComponentState {
+    let mut state = ERC20_COMPONENT_STATE();
+    let mut mock_state = ERC20VOTES_CONTRACT_STATE();
+
+    // Mint SUPPLY tokens to DELEGATOR
+    mock_state.erc20.mint(DELEGATOR(), SUPPLY);
+    state
+}
+
+fn setup_account(public_key: felt252) -> ContractAddress {
+    let mut calldata = array![public_key];
+    utils::declare_and_deploy("SnakeAccountMock", calldata)
+}
+
+//
+// Common tests for Votes
+//
+
+#[test]
+fn test_get_votes() {
+    let mut state = setup_erc721_votes();
+    start_cheat_caller_address(test_address(), DELEGATOR());
+    // Before delegating, the DELEGATOR has 0 votes
+    assert_eq!(state.get_votes(DELEGATOR()), 0);
+    state.delegate(DELEGATOR());
+
+    assert_eq!(state.get_votes(DELEGATOR()), ERC_721_INITIAL_MINT);
+}
+
+#[test]
+fn test_get_past_votes() {
+    let mut state = setup_erc721_votes();
+    let mut trace = state.Votes_delegate_checkpoints.entry(DELEGATOR());
+
+    start_cheat_block_timestamp_global('ts10');
+
+    trace.push('ts1', 3);
+    trace.push('ts2', 5);
+    trace.push('ts3', 7);
+
+    assert_eq!(state.get_past_votes(DELEGATOR(), 'ts2'), 5);
+    assert_eq!(state.get_past_votes(DELEGATOR(), 'ts5'), 7);
+}
+
+#[test]
+#[should_panic(expected: ('Votes: future Lookup',))]
+fn test_get_past_votes_future_lookup() {
+    let state = setup_erc721_votes();
+
+    start_cheat_block_timestamp_global('ts1');
+    state.get_past_votes(DELEGATOR(), 'ts2');
+}
+
+#[test]
+fn test_get_past_total_supply() {
+    let mut state = setup_erc721_votes();
+    let mut trace = state.Votes_total_checkpoints.deref();
+
+    start_cheat_block_timestamp_global('ts10');
+    trace.push('ts1', 3);
+    trace.push('ts2', 5);
+    trace.push('ts3', 7);
+
+    assert_eq!(state.get_past_total_supply('ts2'), 5);
+    assert_eq!(state.get_past_total_supply('ts5'), 7);
+}
+
+#[test]
+#[should_panic(expected: ('Votes: future Lookup',))]
+fn test_get_past_total_supply_future_lookup() {
+    let state = setup_erc721_votes();
+    start_cheat_block_timestamp_global('ts1');
+    state.get_past_total_supply('ts2');
+}
+
+#[test]
+fn test_self_delegate() {
+    let mut state = setup_erc721_votes();
+    let contract_address = test_address();
+    let mut spy = spy_events();
+    start_cheat_caller_address(contract_address, DELEGATOR());
+
+    state.delegate(DELEGATOR());
+    spy.assert_event_delegate_changed(contract_address, DELEGATOR(), ZERO(), DELEGATOR());
+    spy
+        .assert_only_event_delegate_votes_changed(
+            contract_address, DELEGATOR(), 0, ERC_721_INITIAL_MINT
+        );
+    assert_eq!(state.get_votes(DELEGATOR()), ERC_721_INITIAL_MINT);
+}
+
+#[test]
+fn test_delegate_to_recipient_updates_votes() {
```

Comment:
Were these added? Forgive me if I'm missing them

Reply:
I just finished those and did some slight improvements to the test suite(since it was getting pretty long) to make it more readable.
You can check this [commit](https://github.com/OpenZeppelin/cairo-contracts/pull/1114/commits/879ced3c99a4668751a5d4d0b57bcb287b4b6cab)